packages/slice-machine/components/Simulator/components/IframeRenderer/index.tsx

```diff
@@ -95,63 +93,66 @@ const IframeRenderer: React.FunctionComponent<IframeRendererProps> = ({
+    height: 0,
+    width: 0,
+  });
+  const viewportRef = useElementSize(({ blockSize, inlineSize }) => {
+    setViewportSize({ height: blockSize, width: inlineSize });
+  }, []);
+
   return (
-    <Box
+    <Flex
+      ref={viewportRef}
       sx={{
-        height: "100%",
-        backgroundColor: "white",
+        alignItems: "center",
+        backgroundColor: "headSection",
+        backgroundImage: "url(/pattern.png)",
+        backgroundRepeat: "repeat",
+        backgroundSize: "10px",
         border: (t) => `1px solid ${String(t.colors?.darkBorder)}`,
         borderRadius: 8,
-        overflow: "hidden",
-        ...(dryRun
-          ? {
-              position: "absolute",
-              top: "0",
-              width: "0",
-              height: "0",
-            }
-          : {}),
-        ...sx,
+        display: "flex",
+        height: "100%",
+        justifyContent: "center",
+        overflowY: "hidden",
+        ...(dryRun ? { display: "none" } : {}),
       }}
     >
-      <Flex
-        sx={{
-          backgroundImage: "url(/pattern.png)",
-          backgroundColor: "headSection",
-          backgroundRepeat: "repeat",
-          backgroundSize: "10px",
-          mx: "auto",
-          flexDirection: "column",
-          justifyContent: "center",
-          overflow: "auto",
-        }}
-      >
-        <Flex
-          sx={{
-            justifyContent: "center",
-            margin: "0 auto",
-            alignContent: "center",
-            width: screenDimensions.width,
-            height: screenDimensions.height,
+      {simulatorUrl ? (
+        <iframe
+          id="__iframe-renderer"
+          ref={iframeRef}
+          src={simulatorUrl}
+          style={{
+            border: "none",
+            maxHeight: `${iframeSize.height}px`,
+            maxWidth: `${iframeSize.width}px`,
+            minHeight: `${iframeSize.height}px`,
+            minWidth: `${iframeSize.width}px`,
+            overflowY: "auto",
+            transform: `scale(${getScaling(iframeSize, viewportSize)})`,
           }}
-        >
-          {client?.connected ? <div id="__iframe-ready" /> : null}
-          {simulatorUrl ? (
-            <iframe
-              id="__iframe-renderer"
-              ref={refCallback}
-              src={simulatorUrl}
-              style={{
-                border: "none",
-                width: "100%",
-                height: "100%",
-              }}
-            />
-          ) : null}
-        </Flex>
-      </Flex>
-    </Box>
+        />
+      ) : null}
+      {client?.connected ? <div id="__iframe-ready" /> : null}
+    </Flex>
   );
 };
 
+function getScaling(
+  { height: iframeHeight, width: iframeWidth }: ScreenDimensions,
+  { height: viewportHeight, width: viewportWidth }: ScreenDimensions
+): number {
+  const widthDiff = iframeWidth - viewportWidth;
+  const heightDiff = iframeHeight - viewportHeight;
+  if (iframeWidth > viewportWidth && widthDiff > heightDiff) {
```

Comment:
What if both conditions are met (the wanted height AND width is bigger than the box)
Why do we scale only by one of them ?

Reply:
It is crystal clear with explanations.
The fact that is very clear with few explanations is an excellent argument for tests / comments